tests/pubsub/check_pubsub_subscribe.c

```diff
@@ -0,0 +1,451 @@
+        memset(&readerGroupConfig, 0, sizeof(readerGroupConfig));
+        readerGroupConfig.name = UA_STRING("readerGroup 1");
+        retVal |= UA_PubSubConnection_addReaderGroup(server, connection1, &readerGroupConfig, &localreaderGroup);
+        ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
+        memset(&dataSetreaderConfig, 0, sizeof(dataSetreaderConfig));
+        retVal |= UA_PubSubReaderGroup_addDataSetReader(server, localreaderGroup, &dataSetreaderConfig, &localDataSetreader);
+        ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
+        retVal |= UA_PubSubDataSetReader_addTargetVariables(NULL, &localreaderGroup, localDataSetreader, UA_PUBSUB_SDS_TARGET);
+        ck_assert_int_ne(retVal, UA_STATUSCODE_GOOD);
+        UA_PubSubConnection_removeReaderGroup(server, localreaderGroup);
+    } END_TEST
+
+int main(void) {
+    TCase *tc_add_pubsub_readergroup = tcase_create("PubSub readerGroup items handling");
+    tcase_add_checked_fixture(tc_add_pubsub_readergroup, setup, teardown);
+
+    //Test cases for ReaderGroup functionality
+    tcase_add_test(tc_add_pubsub_readergroup, AddReaderGroupWithValidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, AddReaderGroupWithNullConfig);
+    tcase_add_test(tc_add_pubsub_readergroup, AddReaderGroupWithInvalidConnectionId);
+    tcase_add_test(tc_add_pubsub_readergroup, RemoveReaderGroupWithInvalidIdentifier);
+    tcase_add_test(tc_add_pubsub_readergroup, UpdateReaderGroupWithInvalidIdentifier);
+    tcase_add_test(tc_add_pubsub_readergroup, GetReaderGroupConfigWithInvalidConfig);
+    tcase_add_test(tc_add_pubsub_readergroup, GetReaderGroupConfigWithInvalidIdentifier);
+    tcase_add_test(tc_add_pubsub_readergroup, GetReaderGroupConfigWithValidConfig);
+
+    //Test cases for DataSetReader functionality
+    tcase_add_test(tc_add_pubsub_readergroup, AddDataSetReaderWithValidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, AddDataSetReaderWithNullConfig);
+    tcase_add_test(tc_add_pubsub_readergroup, RemoveDataSetReaderWithValidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, RemoveDataSetReaderWithInvalidIdentifier);
+    tcase_add_test(tc_add_pubsub_readergroup, UpdateDataSetReaderConfigWithInvalidConfig);
+    tcase_add_test(tc_add_pubsub_readergroup, UpdateDataSetReaderConfigWithInvalidId);
+    tcase_add_test(tc_add_pubsub_readergroup, UpdateDataSetReaderConfigWithValidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, GetDataSetReaderConfigWithValidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, GetDataSetReaderConfigWithInvalidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, GetDataSetReaderConfigWithInvalidIdentifier);
+    tcase_add_test(tc_add_pubsub_readergroup, CreateTargetVariableWithInvalidConfiguration);
+    tcase_add_test(tc_add_pubsub_readergroup, AddTargetVariableWithInvalidConfiguration);
+
+    Suite *suite = suite_create("PubSub readerGroups/reader/Fields handling and publishing");
```

Comment:
Nice tests!
It would be perfect if eventually we could have a test also for the receiving.
So publisher and subscriber working together.